src/Dofs/ConstraintHandler.jl

```diff
@@ -154,6 +154,52 @@ function _add!(ch::ConstraintHandler, dbc::Dirichlet, interpolation::Interpolati
     append!(ch.prescribed_dofs, constrained_dofs)
 end
 
+function _add!(ch::ConstraintHandler, dbc::Dirichlet, bcnodes::Set{Int}, interpolation::Interpolation, field_dim::Int, offset::Int)
+    if interpolation !== default_interpolation(getcelltype(ch.dh.grid))
+        warn("adding constraint to nodeset is not recommended for sub/super-parametric approximations.")
+    end
+
+    ncomps = length(dbc.components)
+    nnodes = getnnodes(ch.dh.grid)
+    interpol_points = ndofs_per_cell(ch.dh)
+    _celldofs = fill(0, ndofs_per_cell(ch.dh))
+    node_dofs = zeros(Int, ncomps, nnodes)
+    visited = BitVector(nnodes)
+    visited .= false
+    for (cellidx, cell) in enumerate(ch.dh.grid.cells)
+        celldofs!(_celldofs, ch.dh, cellidx) # update the dofs for this cell
+        for idx in 1:min(interpol_points, length(cell.nodes))
+            node = cell.nodes[idx]
+            if !visited[node]
+                noderange = (offset + (idx-1)*field_dim + 1):(offset + idx*field_dim) # the dofs in this node
+                for (i,c) in enumerate(dbc.components)
+                    node_dofs[i,node] = _celldofs[noderange[c]]
+                    @debug println("adding dof $(_celldofs[noderange[c]]) to node_dofs")
+                end
+                visited[node] = true
+            end
+        end
+    end
+
+    constrained_dofs = Int[]
+    sizehint!(constrained_dofs, ncomps*length(bcnodes))
+    sizehint!(dbc.local_face_dofs, length(bcnodes))
+    for node in bcnodes
+        if !visited[node]
+            throw("Unable to add a Dirichlet boundary condition to node $node as there are no degrees of freedom on this node.")
```

Comment:
`throw` -> `error`

Reply:
Sure.

Reply:
We should throw an exception not a String :)

Reply:
True, but this works:
```julia
julia> throw("eee")
ERROR: "eee"
```